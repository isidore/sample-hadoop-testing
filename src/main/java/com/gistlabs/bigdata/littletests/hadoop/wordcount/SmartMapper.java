package com.gistlabs.bigdata.littletests.hadoop.wordcount;

import org.apache.hadoop.mapreduce.Mapper;

public abstract class SmartMapper<KeyIn, ValueIn, KeyOut, ValueOut>
    extends
      Mapper<KeyIn, ValueIn, KeyOut, ValueOut> implements Transform<KeyIn, ValueIn, KeyOut, ValueOut>
{
}
