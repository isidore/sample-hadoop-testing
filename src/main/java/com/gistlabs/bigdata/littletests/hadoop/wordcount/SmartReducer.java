package com.gistlabs.bigdata.littletests.hadoop.wordcount;

import org.apache.hadoop.mapreduce.Reducer;

public abstract class SmartReducer<KeyIn, ValueIn, KeyOut, ValueOut>
    extends
      Reducer<KeyIn, ValueIn, KeyOut, ValueOut> implements Transform<KeyIn, ValueIn, KeyOut, ValueOut>
{
}