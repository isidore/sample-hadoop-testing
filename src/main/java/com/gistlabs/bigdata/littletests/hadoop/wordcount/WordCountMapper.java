package com.gistlabs.bigdata.littletests.hadoop.wordcount;

import java.io.IOException;

import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;

public class WordCountMapper extends SmartMapper<LongWritable, Text, Text, LongWritable>
{
  @Override
  protected void map(LongWritable offset, Text text, Context context) throws IOException, InterruptedException
  {
    for (String token : text.toString().split("\\s+"))
    {
      context.write(new Text(token), new LongWritable(1));
    }
  }
  @Override
  public Class<LongWritable> getKeyInType()
  {
    return LongWritable.class;
  }
  @Override
  public Class<Text> getValueInType()
  {
    return Text.class;
  }
  @Override
  public Class<Text> getKeyOutType()
  {
    return Text.class;
  }
  @Override
  public Class<LongWritable> getValueOutType()
  {
    return LongWritable.class;
  }
}