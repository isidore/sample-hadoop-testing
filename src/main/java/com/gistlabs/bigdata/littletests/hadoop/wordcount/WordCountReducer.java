package com.gistlabs.bigdata.littletests.hadoop.wordcount;

import java.io.IOException;

import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;

public class WordCountReducer extends SmartReducer<Text, LongWritable, Text, LongWritable>
{
  @Override
  protected void reduce(Text token, Iterable<LongWritable> counts, Context context) throws IOException,
      InterruptedException
  {
    long n = 0;
    for (LongWritable count : counts)
      n += count.get();
    context.write(token, new LongWritable(n));
  }
  @Override
  public Class<Text> getKeyInType()
  {
    return Text.class;
  }
  @Override
  public Class<LongWritable> getValueInType()
  {
    return LongWritable.class;
  }
  @Override
  public Class<Text> getKeyOutType()
  {
    return Text.class;
  }
  @Override
  public Class<LongWritable> getValueOutType()
  {
    return LongWritable.class;
  }
}