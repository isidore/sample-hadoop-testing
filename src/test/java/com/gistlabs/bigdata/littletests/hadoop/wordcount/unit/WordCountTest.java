package com.gistlabs.bigdata.littletests.hadoop.wordcount.unit;

import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

import com.gistlabs.bigdata.littletests.hadoop.wordcount.WordCountMapper;
import com.gistlabs.bigdata.littletests.hadoop.wordcount.WordCountReducer;

@UseReporter(DiffReporter.class)
public class WordCountTest// extends AbstractWordCount
{
  @Test
  public void testMapper() throws Exception
  {
    HadoopApprovals.verifyMapping(new WordCountMapper(), 0, "cat cat dog");
  }
  @Test
  public void testReducer() throws Exception
  {
    HadoopApprovals.verifyReducer(new WordCountReducer(), "Smiles", 3, 4);
  }
  @Test
  public void testMapReduce() throws Exception
  {
    HadoopApprovals.verifyMapReduce(new WordCountMapper(), new WordCountReducer(), 0, "cat cat dog");
  }
}
