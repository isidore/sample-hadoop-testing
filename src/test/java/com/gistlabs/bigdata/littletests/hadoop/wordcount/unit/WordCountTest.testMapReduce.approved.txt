[cat cat dog]

 -> maps via WordCountMapper to -> 

(cat, 1)
(cat, 1)
(dog, 1)


 -> reduces via WordCountReducer to -> 

(cat, 2)
(dog, 1)
